plenum/server/batch_handlers/audit_batch_handler.py

```diff
@@ -0,0 +1,93 @@
+from ledger.ledger import Ledger
+from plenum.common.constants import AUDIT_LEDGER_ID, TXN_VERSION, AUDIT_TXN_VIEW_NO, AUDIT_TXN_PP_SEQ_NO, \
+    AUDIT_TXN_LEDGERS_SIZE, AUDIT_TXN_LEDGER_ROOT, AUDIT_TXN_STATE_ROOT
+from plenum.common.ledger_uncommitted_tracker import LedgerUncommittedTracker
+from plenum.common.transactions import PlenumTransactions
+from plenum.common.txn_util import init_empty_txn, set_payload_data, get_payload_data, get_seq_no
+from plenum.server.batch_handlers.batch_request_handler import BatchRequestHandler
+from plenum.server.batch_handlers.three_pc_batch import ThreePcBatch
+from plenum.server.database_manager import DatabaseManager
+
+
+class AuditBatchHandler(BatchRequestHandler):
+
+    def __init__(self, database_manager: DatabaseManager):
+        super().__init__(database_manager, AUDIT_LEDGER_ID)
+        # TODO: move it to BatchRequestHandler
+        self.tracker = LedgerUncommittedTracker(None, self.ledger.size)
+
+    def post_batch_applied(self, three_pc_batch: ThreePcBatch, prev_handler_result=None):
+        self._add_to_ledger(three_pc_batch)
+        self.tracker.apply_batch(None, self.ledger.uncommitted_size)
+
+    def post_batch_rejected(self, ledger_id, prev_handler_result=None):
+        _, txn_count = self.tracker.reject_batch()
+        self.ledger.discardTxns(txn_count)
+
+    def commit_batch(self, ledger_id, txn_count, state_root, txn_root, pp_time, prev_result=None):
+        _, txns_count = self.tracker.commit_batch()
+        _, committedTxns = self.ledger.commitTxns(txns_count)
+        return committedTxns
+
+    def _add_to_ledger(self, three_pc_batch: ThreePcBatch):
+        # if PRE-PREPARE doesn't have audit txn (probably old code) - do nothing
+        # TODO: remove this check after all nodes support audit ledger
+        if not three_pc_batch.has_audit_txn:
+            return
```

Comment:
Please have a look at the TODO comment above. 
We have to do this until the first (non-forced) update of the Main Net is done, otherwise some nodes during update may write to audit ledger, and some not, which will lead to inconsistent state or view changes.